pkg/policy/portrange_test.go

```diff
@@ -0,0 +1,347 @@
+				{0b0000000000001010, 0b1111111111111111}, // 10
+			},
+		},
+		{
+			start: 0,
+			end:   16,
+			expected: []MaskedPort{
+				{0b0000000000000000, 0b1111111111110000}, // 0xxxx
+				{0b0000000000010000, 0b1111111111111111}, // 10000
+			},
+		},
+		{
+			start: 0b0000000000010000, // 16
+			end:   0b0000000110000111, // 391
+			expected: []MaskedPort{
+				{0b0000000000010000, 0b1111111111110000}, // 00001xxxx, 16-31
+				{0b0000000000100000, 0b1111111111100000}, // 0001xxxxx, 32-63
+				{0b0000000001000000, 0b1111111111000000}, // 001xxxxxx, 64-127
+				{0b0000000010000000, 0b1111111110000000}, // 01xxxxxxx, 128-255
+				{0b0000000100000000, 0b1111111110000000}, // 01xxxxxxx, 256-383
+				{0b0000000110000000, 0b1111111111111000}, // 10000000x, 384-391
+			},
+		},
+		{
+			start: 22,
+			end:   23,
+			expected: []MaskedPort{
+				{0b0000000000010110, 0b1111111111111110}, // 22-23
+			},
+		},
+		{
+			start: 23,
+			end:   24,
+			expected: []MaskedPort{
+				{0b0000000000010111, 0b1111111111111111}, // 23
+				{0b0000000000011000, 0b1111111111111111}, // 24
+			},
+		},
+		{
+			start: 0,
+			end:   0x7fff,
+			expected: []MaskedPort{
+				{port: 0x0, mask: 0x8000}, // 0-0x7fff
+			},
+		},
+		{
+			start: 256,
+			end:   256,
+			expected: []MaskedPort{
+				{port: 0x100, mask: 0xffff},
+			},
+		},
+		{
+			start: 65535,
+			end:   65535,
+			expected: []MaskedPort{
+				{port: 65535, mask: 0xffff},
+			},
+		},
+		{
+			start: 32767,
+			end:   32768,
+			expected: []MaskedPort{
+				{port: 0x7fff, mask: 0xffff},
+				{port: 0x8000, mask: 0xffff},
+			},
+		},
+		{
+			start: 0b0101010101010101, // 0x5555
+			end:   0b0101010111010101, // 0x55d5
+			expected: []MaskedPort{
+				{port: 0b0101010101010101, mask: 0b1111111111111111},
+				{port: 0b0101010101010110, mask: 0b1111111111111110},
+				{port: 0b0101010101011000, mask: 0b1111111111111000},
+				{port: 0b0101010101100000, mask: 0b1111111111100000},
+				{port: 0b0101010110000000, mask: 0b1111111111000000},
+				{port: 0b0101010111000000, mask: 0b1111111111110000},
+				{port: 0b0101010111010000, mask: 0b1111111111111100},
+				{port: 0b0101010111010100, mask: 0b1111111111111110},
+			},
+		},
+		// This is all ports.
+		{
+			start: 0,
+			end:   0,
+			expected: []MaskedPort{
+				{port: 0, mask: 0},
+			},
+		},
+		// This is too.
+		{
+			start: 0,
+			end:   65535,
+			expected: []MaskedPort{
+				{port: 0, mask: 0},
+			},
+		},
+		// This is valid, as "0" defines no range.
+		{
+			start: 65535,
+			end:   0,
+			expected: []MaskedPort{
+				{port: 0xffff, mask: 0xffff}, // 65535
+			},
+		},
+		// These remaining tests are testing invalid cases where start >= end.
+		// For now, these cases return the start port with a full mask,
+		// indicating that only the start port should be part of the range.
+		// This is technically correct for the case of end port being "0"
+		// and the value of the port, but is ambiguous otherwise.
+		{
```

Comment:
What's the behavior if endport > 65535 but start is valid. can we also add a test for port range something like 50000-70000?